route/src/route_generator_worker.cpp

```diff
@@ -418,6 +424,20 @@ namespace route {
     
     bool RouteGeneratorWorker::spin_callback()
     {
+        if(localFunction()==true)
+        {
+           this->rs_worker_.on_route_event(RouteStateWorker::RouteEvent::ROUTE_INVALIDATION);
+           publish_route_event(cav_msgs::RouteEvent::ROUTE_INVALIDATION);
+           
+           auto route=routing(destination_points_in_map_.front(),
```

Comment:
On a basic level this is fine, but I think there is an edge case we need to consider. 
The route file is made up of a list of destination points. If the vehicle, has already passed one of these points it will cause the new reroute to loop back on itself. I would suggest adding some logic to this node which tracks which points have been passed and then removes them from the destination_points_in_map_ list. 

I think the way to approach this would be to compute the corresponding route downtrack distances for each point after the original route it generated, then you could compare the output vehicle downtrack from this node against those points on each spin loop to remove points as needed.